BucketDevice.h sign comparison warning (clang)

Buidling with clang as the host compiler highlighted the following warning (not emitted by msvc/ gcc even though sign comparion warnings are enabled afaik). 

```
BucketDevice.h:309:20: warning: comparison of integers of different signs: 'const unsigned int' and 'const flamegpu::IntT' (aka 'const int') [-Wsign-compare]
```

Not super urgent, but worth remembering to resolve.

Also highlighted missing `override` statements which i've enabled in an ongoing PR.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 ### Changed
++ Use `IntT` in `MessageBucketDevice` to resolve clang sign comparison warnings ([#554](https://github.com/FLAMEGPU/FLAMEGPU2/issues/554))
 
 ### Deprecated
 

--- a/include/flamegpu/runtime/messaging/MessageBucket/MessageBucketDevice.cuh
+++ b/include/flamegpu/runtime/messaging/MessageBucket/MessageBucketDevice.cuh
@@ -39,14 +39,14 @@ class MessageBucket::In {
             /**
             * This is the index of the currently accessed message, relative to the full message list
             */
-            unsigned int cell_index;
+            IntT cell_index;
 
          public:
             /**
             * Constructs a message and directly initialises all of it's member variables
             * @note See member variable documentation for their purposes
             */
-            __device__ Message(const Filter &parent, const unsigned int &_cell_index)
+            __device__ Message(const Filter &parent, const IntT &_cell_index)
                 : _parent(parent)
                 , cell_index(_cell_index) { }
             /**
@@ -106,7 +106,7 @@ class MessageBucket::In {
             * This iterator is constructed by MessageBucket::In::Filter::begin()(IntT)
             * @see MessageBucket::In::Operator()(IntT)
             */
-            __device__ iterator(const Filter &parent, const unsigned int &cell_index)
+            __device__ iterator(const Filter &parent, const IntT &cell_index)
                 : _message(parent, cell_index) {
                 // Increment to find first message
                 ++_message;
@@ -171,7 +171,7 @@ class MessageBucket::In {
         /**
         * Returns the number of messages in the filtered bucket
         */
-        inline __device__ unsigned int size(void) const {
+        inline __device__ IntT size(void) const {
             return bucket_end - bucket_begin;
         }
 

